File: compiloop.py
#!/usr/bin/python
# Wait for changes to relevant files, then compile and run test. And
# do this forever.

import os, sys, time

filenames = ["sqlite3_bindings.cc", "wscript", "sqlite.js", "test.js"]

mtime = []
while True:
  m = [os.stat(filename).st_mtime for filename in filenames]
  if mtime != m:
    os.system("clear; rm -f test.db")
    os.system("node-waf build && node test.js && sleep 1 && sqlite3 test.db .dump");
    mtime = m

  time.sleep(1)

File: compiloop.js
var sys = require("sys");

function handler(curr, prev) {
  sys.puts("Handling");
  sys.puts("the current mtime is: " + curr.mtime);
  sys.puts("the previous mtime was: " + prev.mtime);
  sys.exec("clear;rm -f test.db; node-waf build && node test.js");
}

sys.puts(JSON.stringify(process.ARGV));
for (f in process.ARGV) {
  f = process.ARGV[f]
  sys.puts("Watching " + f);
  process.watchFile(f, handler);
}

//var tcp = require("tcp");
//var server = tcp.createServer();
//server.listen(7000, "localhost");

for (;;) {
  sys.exec("sleep 1").wait();
}

//var p = new process.Promise();
//p.wait(); 